src/main/java/com/nextstep/javaRacing/StringAddCalculator.java

```diff
@@ -0,0 +1,40 @@
+package com.nextstep.javaRacing;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final Pattern CUSTOM_DELIMITER_PATTERN = Pattern.compile("//(.)\n(.*)");
+
+    public static int splitAndSum(String s) {
+        if (s == null || s.isEmpty()) return 0;
+        String[] tokens = split(s);
+        return sum(tokens);
+    }
+
+    private static String[] split(String s) {
+        Matcher matcher = CUSTOM_DELIMITER_PATTERN.matcher(s);
+        if (matcher.find()) {
+            String customDelimiter = matcher.group(1);
+            String valuePattern = matcher.group(2);
+            return valuePattern.split(customDelimiter);
+        }
+        return s.split("[,:]");
+    }
+
+    private static int sum(String[] tokens) {
+        return Arrays.stream(tokens).mapToInt(s->parse(s)).sum();
```

Comment:
체이닝 메서드의 경우 주로 `.` 단위로 개행을 해줍니다. 그렇다면 가독성이 좋은 코드를 유지할 수 있습니다 :)

```suggestion
        return Arrays.stream(tokens)
                .mapToInt(StringAddCalculator::parse)
                .sum();
```

추가로 제가 IDE를 켜서 확인해보지는 않았는데 `mapToInt()`에서 함수의 형식이 인자와 일치한다면 참조 타입으로 함수를 넘겨줄 수 있습니다.